integration/monitor/rate-limiter.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * A simple rate limiter that allows a limited number of leases in a window of time.
+ * @constructor
+ * @param {number} window - The window of time in milliseconds in which to count leases.
+ * @param {number} limit - The maximum number of leases in the window. Default 1.
+ */
+class RateLimiter {
```

Comment:
📓 I looked at a few other rate limiters. But, they were designed more for rate limiting requests to a web app. One kept all the requests in a queue for later execution. That isn't what we want. We just want to discard if over the limit.